File: python/encryption/block_and_stream_cipher.py
# block_and_stream_cipher.py

import os
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes

# AES Block Cipher Encryption
def aes_encrypt(text: str) -> dict:
    key = os.urandom(32)
    iv = os.urandom(16)
    cipher = Cipher(algorithms.AES(key), modes.CFB(iv))
    encryptor = cipher.encryptor()
    encrypted = encryptor.update(text.encode()) + encryptor.finalize()
    return {'encrypted': encrypted, 'key': key, 'iv': iv}

# AES Decrypt
def aes_decrypt(data: dict) -> str:
    cipher = Cipher(algorithms.AES(data['key']), modes.CFB(data['iv']))
    decryptor = cipher.decryptor()
    decrypted = decryptor.update(data['encrypted']) + decryptor.finalize()
    return decrypted.decode()

if __name__ == "__main__":
    msg = "Hello, World!"
    encrypted_data = aes_encrypt(msg)
    print(f"Encrypted (AES): {encrypted_data}")
    print(f"Decrypted (AES): {aes_decrypt(encrypted_data)}")

File: js/encryption/blockAndStreamCipher.js
// blockAndStreamCipher.js

const crypto = require('crypto');

// AES as a block cipher
function aesEncrypt(text) {
  const key = crypto.randomBytes(32);
  const iv = crypto.randomBytes(16);
  const cipher = crypto.createCipheriv('aes-256-cbc', key, iv);
  let encrypted = cipher.update(text, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  return { encrypted, key: key.toString('hex'), iv: iv.toString('hex') };
}

// RC4 as a stream cipher
function rc4Encrypt(text, key) {
  const cipher = crypto.createCipher('rc4', key);
  let encrypted = cipher.update(text, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  return encrypted;
}

module.exports = { aesEncrypt, rc4Encrypt };
